src/core/BufferAttribute.js

```diff
@@ -298,6 +298,26 @@ THREE.BufferAttribute.prototype = {
 
 		return new this.constructor().copy( this );
 
+	},
+
+	merge: function( attribute ) {
+
+		var result, first = this.array, second = attribute.array, length = first.length;
+
+		if ( first.constructor !== second.constructor || this.itemSize !== attribute.itemSize ) {
+
+			console.error( 'THREE.BufferAttribute.merge(): type mismatch', attribute );
+
+		}
+
+		result = new first.constructor( length + second.length );
```

Comment:
What if two attributes together will overflow Uint16Array?
Maybe you should do something like this:
`result = new ( ( length + second.length ) > 65535 ? Uint32Array : Uint16Array )( length + second.length );`

Reply:
Uint16Array values are capped at 65535, not their length, so this should be fine.